MinimumRSA.java

```diff
@@ -0,0 +1,37 @@
+public class MinimumRSA {
+    public int findMin(int[] nums) {
+        return search(nums,0,nums.length-1);
+    }
+    public int search(int[] nums,int start,int end){
+        if(start>end)
+            return -1;
+        int mid = start+(end-start)/2;
+        if(nums[mid]<=nums[start]&& nums[mid]<=nums[end]){
+            if(end-start<=2)
+                return nums[mid];
+            else{
+                start = start+1;
+                end = end - 1;
+                return search(nums,start,end);
```

Comment:
Can you do this iteratively? Which solution is better recursive or iterative?

Reply:
Implemented using iterative approach (MinimumRSA2.java). 
In my opinion, both have the same time complexity. But if memory is limited iterative approach is better because for large arrays, using a recursive approach takes more space and may cause StackOverflow .

Reply:
Thats the great explain for iterative vs recursive approach Bharath